packages/components/src/components/expansion-panel/expansion-panel.jsx

```diff
@@ -3,34 +3,23 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import Icon from '../icon';
 
-class ExpansionPanel extends React.Component {
-    state = {
-        open_ids: [],
-        is_open: false,
-    };
+const ExpansionPanel = ({ message }) => {
+    const [open_ids, setOpenIds] = React.useState([]);
+    const [is_open, setIsOpen] = React.useState(false);
 
-    onClick = () => {
+    const onClick = () => {
         // close if clicking the expansion panel that's open, otherwise open the new one
-        this.setState({ is_open: !this.state.is_open });
+        setIsOpen(!is_open);
     };
 
-    onArrayItemClick = id => {
-        this.setState(state => {
-            if (state.open_ids.includes(id)) {
-                const open_ids = state.open_ids.filter(open_id => open_id !== id);
-                return {
-                    ...state,
-                    open_ids,
-                };
-            }
-            return {
-                ...state,
-                open_ids: [...state.open_ids, id],
-            };
-        });
+    const onArrayItemClick = id => {
+        if (open_ids.includes(id)) {
+            setOpenIds(open_ids.filter(open_id => open_id !== id));
+        }
+        setOpenIds([...open_ids, id]);
```

Comment:
This behavior is not the same as the original one. if `open_ids.includes(id)` is truthy, then, we are going to : 
filter out the `id` from `open_ids`, and then adding it again via `setOpenIds([...open_ids, id])`.

Reply:
Will `else` statement on line 19 fix the behavior or is there anything missing from original one?